src/combobox/ComboBox.ts

```diff
@@ -164,87 +163,92 @@ export default class ComboBox extends ComboBoxBase<ComboBoxProperties> {
+				if (!this._open && !disabled && !readOnly) {
+					this._openMenu();
+				}
+				else if (this._open) {
+					this._moveActiveIndex(Operation.increase);
+				}
 				break;
 			case Keys.Escape:
 				this._open && this._closeMenu();
 				break;
 			case Keys.Enter:
+			case Keys.Space:
 				if (this._open && results.length > 0) {
-					this._activeIndex === undefined ? this._closeMenu() : this._selectIndex(this._activeIndex);
+					if (isResultDisabled(results[this._activeIndex])) {
+						return;
+					}
+					this._selectIndex(this._activeIndex);
 				}
 				break;
+			case Keys.Home:
+				this._activeIndex = 0;
+				this.invalidate();
+				break;
+			case Keys.End:
+				this._activeIndex = results.length - 1;
+				this.invalidate();
+				break;
 		}
 	}
 
 	private _onMenuChange() {
-		const { onMenuChange } = this.properties;
+		const { key = '', onMenuChange } = this.properties;
 
 		if (!onMenuChange) {
 			return;
 		}
 
-		this._open && !this._wasOpen && onMenuChange(true);
-		!this._open && this._wasOpen && onMenuChange(false);
-	}
-
-	private _onResultMouseDown() {
-		// Maintain underlying input focus on next render
-		this._ignoreBlur = true;
+		this._open && !this._wasOpen && onMenuChange(true, key);
+		!this._open && this._wasOpen && onMenuChange(false, key);
 	}
 
-	private _onResultMouseEnter(event: MouseEvent, index: number) {
-		if (this._isIndexDisabled(index)) {
-			return;
-		}
-
-		this._activeIndex = index;
+	private _onResultHover(): void {
+		this._menuHasVisualFocus = false;
 		this.invalidate();
 	}
 
-	private _onResultMouseUp(event: MouseEvent, index: number) {
-		if (!this._isIndexDisabled(index)) {
-			this._ignoreFocus = true;
-			this._selectIndex(index);
-		}
+	private _onResultMouseDown() {
+		// Maintain underlying input focus on next render
+		this._ignoreBlur = true;
 	}
 
 	private _openMenu() {
-		const { onRequestResults } = this.properties;
+		const {
+			key = '',
+			onRequestResults
+		} = this.properties;
 
-		this._activeIndex = undefined;
+		this._activeIndex = 0;
 		this._open = true;
-		this._focused = true;
-		onRequestResults && onRequestResults(this._inputElement.value);
+		onRequestResults && onRequestResults(key);
+		this.invalidate();
 	}
 
-	private _restoreFocus() {
-		const func = this._focused ? 'focus' : 'blur';
-		this._inputElement[func]();
-	}
-
-	private _scrollIntoView(element: HTMLElement) {
-		const menu = <HTMLElement> element.parentElement;
-		// Scroll menu up so top of highlighted result aligns with top of menu container
-		if (element.offsetTop - menu.scrollTop < 0) {
-			menu.scrollTop = element.offsetTop;
-		}
-		// Scroll menu down so bottom of highlighted result aligns with bottom of menu container
-		else if ((element.offsetTop - menu.scrollTop + element.offsetHeight) > menu.clientHeight) {
-			menu.scrollTop = element.offsetTop - menu.clientHeight + element.offsetHeight;
-		}
-	}
+	// should update to use interesection meta
```

Comment:
Why is this commented out? Is this necessary anymore at all since this looks to be handled in ListBox?

Reply:
ooops, forgot I had that 😂